### PR TITLE
CORE-6130 Schema definition for the DB config

### PR DESCRIPTION
**Pull request by mbrkic-r3**

### CODE REVIEW

File: data/config-schema/src/main/resources/net/corda/schema/configuration/db/1.0/corda.db.json [hunk 1]
```diff
@@ -4,5 +4,56 @@
...
+          "type": "string"
+        },
+        "jdbc": {
+          "description": "JDBC settings",
+          "type": "object",
+          "default": {},
+          "properties": {
+            "driver": {
+              "description": "JDBC driver",
+              "type": "string"
```
Comment (driessamyn) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/db/1.0/corda.db.json]: I think we should have a default for this.

File: data/config-schema/src/main/resources/net/corda/schema/configuration/db/1.0/corda.db.json [hunk 2]
```diff
@@ -4,5 +4,56 @@
...
+            },
+            "directory": {
+              "description": "Directory with JDBC drivers",
+              "type": "string"
```
Comment (driessamyn) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/db/1.0/corda.db.json]: I think we should have a default for this.

File: data/config-schema/src/main/resources/net/corda/schema/configuration/db/1.0/corda.db.json [hunk 3]
```diff
@@ -4,5 +4,56 @@
...
+            }
+          },
+          "additionalProperties": false
+        },
+        "pool": {
+          "description": "Database pool settings",
+          "type": "object",
+          "default": {},
+          "properties": {
+            "max_size": {
```
Comment (driessamyn) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/db/1.0/corda.db.json:49]: I think we should have a default for this.

File: data/config-schema/src/main/resources/net/corda/schema/configuration/db/1.0/corda.db.json [hunk 4]
```diff
@@ -4,5 +4,56 @@
   "title": "Corda Database Configuration Schema",
   "description": "Configuration schema for the database subsection.",
```
Comment (driessamyn) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/db/1.0/corda.db.json]: We should add a note here that this configuration cannot be updated dynamically through the REST endpoint.
Maybe we should even block somebody trying to push this config to the HTTP endpoint? @mnesbit, what do you think?

File: data/config-schema/src/main/resources/net/corda/schema/configuration/db/1.0/corda.db.json (hunk 1 above)
Comment (mbrkic-r3) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/db/1.0/corda.db.json]: set default value to "org.postgresql.Driver"

File: data/config-schema/src/main/resources/net/corda/schema/configuration/db/1.0/corda.db.json (hunk 2 above)
Comment (mbrkic-r3) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/db/1.0/corda.db.json]: set default value to "/opt/jdbc-driver"

File: data/config-schema/src/main/resources/net/corda/schema/configuration/db/1.0/corda.db.json (hunk 3 above)
Comment (mbrkic-r3) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/db/1.0/corda.db.json:49]: set default value to 10

File: data/config-schema/src/main/resources/net/corda/schema/configuration/db/1.0/corda.db.json (hunk 4 above)
Comment (mbrkic-r3) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/db/1.0/corda.db.json]: added note

**Review by driessamyn — DISMISSED** [review]
LGTM

**Review by driessamyn — APPROVED** [review]
LGTM

**Pull request merged**